### ISSUE TITLE
Need reliable way to connect to gemstone with RSR

### ISSUE THREAD

**Issue by ericwinger** [GemTalk/RemoteServiceReplication#16]
Without a reliable way to establish a socket that stays alive between the Dolphin client & the GemStone server, it's not going to be possible to integrate RSR in any meaningful way. The connection shouldn't freeze the gci but should be startable through the gci via Jadeite. 

I've been struggling with this and the best I've come up with is the following: 

First, run this code on the server, 
```
[RsrConnection acceptOn: 4321] fork.
(Delay forSeconds: 1) wait. 
```
Then from the client, 
```
GciSession current executeString: '(Delay forSeconds: 1) wait'.
```
followed by this code which creates the service.

```
| connection | 
connection := RsrConnection connectTo: 4321 on: 'owl'.
service := connection serviceFor: #RowanClassService.
service name: 'Array'. 
service rsrCommand: #update.
service inspect.
```
The difficulty with this approach is that I create many (dozens) of processes on the client and server. 

However, it does allow me to replicate one service at a time without 1) hanging the gci client which forces a Dolphin restart because everything in Jadeite goes through the gci or  2) creating a background process that connects to the server and delays - causes flashing cursor changes among other issues. 

In summary, I need to be able to establish a connection in a reliable manner so I can focus on the actual usage of RSR in Jadeite in order to make reasonable progress toward full integration. 

Here are what I see as the requirements: 

1. RSR must provide an api to establish a connection through Jadeite's gci connection. Creating a separate topaz process to enable replication is a non-starter for a Jadeite application. 
2. The connection should be able to be used repeatedly to replicate services. 
3. Should not hang the gci when not actively replicating. 
4. It should not create many server or client processes. 
5. It should not involve multiple round trips per replication.  (Round trips can be deadly for performance) 

Think that's it. Hopefully,

**Comment by kurtkilpela** [comment]
I don't think this question is relevant anymore. Development no longer happens via old Jadeite but instead via Jadeite for Pharo. That bootstraps a connection via the CGI client and it does cause the CGI to hang until the RSR Connection is closed.
